packages/typespec-client-generator-core/src/package.ts

```diff
@@ -167,7 +241,45 @@ function getSdkPagingServiceMethod<TServiceOperation extends SdkServiceOperation
   });
 }
 
-function getPathFromSegment(context: TCGCContext, type: Model, segments?: string[]): string {
+function getPropertyPathFromModel(
+  context: TCGCContext,
+  model: Model,
+  judge: (property: ModelProperty) => boolean,
+): string | undefined {
```

Comment:
is it possible that we use tcgc type here?
In each generator, we will use this path in those types, therefore it would be more reasonable we get this path from the same source.